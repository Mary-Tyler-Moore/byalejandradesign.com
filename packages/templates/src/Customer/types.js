/** @flow */
import type { Customer as CustomerType } from '@byalejandradesign/checkout-objects';

export type Props = {
  customer: CustomerType,
};
