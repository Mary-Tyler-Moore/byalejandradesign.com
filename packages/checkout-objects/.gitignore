node_modules/
lib/
build/

.rpt2_cache/
