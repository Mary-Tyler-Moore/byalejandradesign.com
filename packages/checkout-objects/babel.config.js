const path = require('path');
const config = require(path.resolve(__dirname, '../../babel.config.js'));

module.exports = (api) => {
  const plugins = [
    [
      '@babel/plugin-proposal-class-properties',
      {
        loose: true,
      },
    ],
    '@babel/plugin-syntax-dynamic-import',
  ];

  return {
    ...config(api),
    plugins,
    babelrcRoots: ['./'],
  };
};
