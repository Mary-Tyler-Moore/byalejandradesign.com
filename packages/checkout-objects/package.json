{
  "name": "@byalejandradesign/checkout-objects",
  "version": "0.0.1",
  "private": true,
  "description": "types and default objects",
  "repository": {
    "type": "git",
    "url": "https://github.com/njmyers/byalejandradesign.com.git"
  },
  "license": "ISC",
  "author": "Nick Myers <nickmyers111@gmail.com>",
  "files": [
    "build"
  ],
  "main": "build/index.cjs.js",
  "module": "build/index.esm.js",
  "scripts": {
    "clean": "shx rm -rf build",
    "prebuild": "run-s clean",
    "build": "rollup -c rollup.config.js",
    "build:flow": "flow gen-flow-files src --out-dir build",
    "develop": "rollup -c rollup.config.js --watch"
  },
  "dependencies": {
    "@babel/runtime": "^7.1.2",
    "smalldash": "^1.1.20",
    "uuid": "^3.3.2"
  },
  "devDependencies": {
    "flow-bin": "^0.85.0",
    "flow-mono-cli": "^1.4.2",
    "flow-typed": "^2.5.1",
    "npm-run-all": "^4.1.3",
    "rollup": "^0.66.6",
    "shx": "^0.3.2"
  }
}
