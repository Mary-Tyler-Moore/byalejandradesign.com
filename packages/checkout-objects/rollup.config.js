import path from 'path';
import resolve from 'rollup-plugin-node-resolve';
import babel from 'rollup-plugin-babel';
import runtimes from '@njmyers/babel-runtime-files';
import pkg from './package.json';

const external = [
  ...Object.keys(pkg.dependencies || {}),
  ...Object.keys(pkg.peerDependencies || {}),
  ...runtimes(),
  'uuid/v1',
];

const plugins = [
  resolve(),
  babel({
    runtimeHelpers: true,
    exclude: 'node_modules/**',
  }),
];

export default [
  {
    input: path.resolve(__dirname, 'src/index.js'),
    external,
    output: {
      file: path.resolve(__dirname, pkg.module),
      format: 'esm',
      sourcemap: true,
    },
    plugins,
  },
  {
    input: path.resolve(__dirname, 'src/index.js'),
    external,
    output: {
      file: path.resolve(__dirname, pkg.main),
      format: 'cjs',
      sourcemap: true,
    },
    plugins,
  },
];
