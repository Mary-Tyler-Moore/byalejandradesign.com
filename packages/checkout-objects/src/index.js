export * from './objects';
export * from './transformers';
export * from './validators';
