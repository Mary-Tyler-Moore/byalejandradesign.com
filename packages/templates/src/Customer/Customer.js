/** @flow */
import * as React from 'react';
import Line from '../Line';
// types
import type { Customer as CustomerType } from '@byalejandradesign/checkout-objects';

export type Props = {
  customer: CustomerType,
};

const Customer = ({ customer }: Props) => (
  <section className="confirmation_customerSection">
    <h4 className="h4-amiri">Customer</h4>
    <Line label="name">{`${customer.firstName} ${customer.lastName}`}</Line>
    <Line label="email">{customer.email}</Line>
    <Line label="phone">{customer.phone}</Line>
  </section>
);

export default Customer;
