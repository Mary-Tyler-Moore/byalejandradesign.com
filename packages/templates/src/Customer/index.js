/** @flow */
export { default } from './Customer';
export * from './Customer';
