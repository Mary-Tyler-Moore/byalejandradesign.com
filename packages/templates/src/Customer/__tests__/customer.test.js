import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Customer from '../Customer';
import { transaction } from '@byalejandradesign/checkout-objects';

describe('it works', () => {
  test('it loads without crashing', () => {
    const render = TestRenderer.create(<Customer />);
    expect(render.toJSON()).toMatchSnapshot();
  });

  test('it renders a customer', () => {
    const render = TestRenderer.create(
      <Customer customer={transaction.customer} />
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
