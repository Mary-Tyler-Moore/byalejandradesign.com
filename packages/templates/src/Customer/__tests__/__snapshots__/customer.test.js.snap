// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works it loads without crashing 1`] = `null`;

exports[`it works it renders a customer 1`] = `
<section
  className="confirmation_customerSection"
>
  <h5
    className="h5-amiri"
  >
    Customer
  </h5>
  <p
    className="body-sourceSans-2"
  >
    <strong>
      Name
      : 
    </strong>
    Nick Myers
  </p>
  <p
    className="body-sourceSans-2"
  >
    <strong>
      Email
      : 
    </strong>
    nick@email.com
  </p>
  <p
    className="body-sourceSans-2"
  >
    <strong>
      Phone
      : 
    </strong>
    212-192-1222
  </p>
</section>
`;
